### PR TITLE
Rename image parameter names

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Here I'm renaming some image parameters, where it appears crucial to me, e.g. if a parameter is a list of coordinates, it should not be called input_image. 

Additionally, I added and corrected documentation in other cases.

partially addresses https://github.com/clEsperanto/pyclesperanto/issues/260

### CODE REVIEW

**Comment by haesleinhuepf** [comment]
Hey @StRigaud I'm running out of time. So here are coming the most important image parameter renamings + documentation upgrads for other functions where it seems important.

**Comment by StRigaud** [comment]
I will have a look at the tests and see to fix it

File: clic/include/tier3.hpp [hunk 1]
```diff
@@ -261,8 +261,8 @@ histogram_func(const Device::Pointer & device,
  * + 1)</pre>
  *
  * @param device Device to perform the operation on. [const Device::Pointer &]
- * @param src0 [const Array::Pointer &]
- * @param src1 [const Array::Pointer &]
+ * @param src0 First binary image to compare [const Array::Pointer &]
+ * @param src1 Second binary image to compare[const Array::Pointer &]
```
Comment (StRigaud) [inline clic/include/tier3.hpp:265]:
```suggestion
 * @param src1 Second binary image to compare [const Array::Pointer &]
```

**Comment by StRigaud** [comment]
FYI, having the label as second parameter to the statistic function is really not convenient :disappointed:

**Comment by StRigaud** [comment]
- clEsperanto/pyclesperanto#273

Works locally, will integrate it with a final release before I2K hopefully

**Pull request merged**